File: ticketProject5-DistributedC/TicketSystemCC/persistance/Databases/BuyerDatabase.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Data;
using persistance.Interfaces;
using model;
using model.Validators;

namespace persistance.Database
{
    public class BuyerDatabase : IBuyerRepository
    {
        //private static readonly ILog log = LogManager.GetLogger("BuyerDatabase");
        private Validator<Buyer> validator;

        public BuyerDatabase(Validator<Buyer> validator)
        {
            //log.Info("Creating BuyerDatabase");
            this.validator = validator;
        }

        public Buyer extractEntity(List<String> attributes)
        {
            if (attributes.Count < 5)
                throw new ValidationException("Insuficient datas!\n");
            Buyer buyer = new Buyer(long.Parse(attributes[1]), attributes[2], attributes[3], long.Parse(attributes[4]))
            {
                Id = Int64.Parse(attributes[0])
            };
            return buyer;
        }

        protected String createEntityAsString(Buyer entity)
        {
            return entity.Id + ";" + entity.IdAR + ";" + entity.FirstName + ";" + entity.LastName + ";" + entity.NoTickets;
        }

        public Buyer FindOne(long id)
        {
            IDbConnection con = DBUtils.getConnection();

            try
            {
                using (var comm = con.CreateCommand())
                {
                    comm.CommandText = "select id, idAR, firstName, lastName, noTickets from Buyers where id=@id";
                    IDbDataParameter paramId = comm.CreateParameter();
                    paramId.ParameterName = "@id";
                    paramId.Value = id;
                    comm.Parameters.Add(paramId);

                    using (var dataR = comm.ExecuteReader())
                    {
                        if (dataR.Read())
                        {
                            long idd = dataR.GetInt32(0);
                            long idAR = dataR.GetInt64(1);
                            String firstName = dataR.GetString(2);
                            String lastName = dataR.GetString(3);
                            long noTickets = dataR.GetInt64(4);

                            Buyer buyer = new Buyer(idAR, firstName, lastName, noTickets)
                            {
                                Id = idd
                            };

                            return buyer;
                        }
                    }
                }
            }
            catch (Exception ex)
            {
                Console.ForegroundColor = ConsoleColor.Red;
                Console.WriteLine("Mesajul erorii este {0}", ex.Message);
            }
            return null;
        }

        public IEnumerable<Buyer> FindAll()
        {
            IDbConnection con = DBUtils.getConnection();
            IList<Buyer> buyers = new List<Buyer>();

            try
            {
                using (var comm = con.CreateCommand())
                {
                    comm.CommandText = "select id, idAR, firstName, lastName, noTickets from Buyers";

                    using (var dataR = comm.ExecuteReader())
                    {
                        while (dataR.Read())
                        {
                            long idd = dataR.GetInt32(0);
                            long idAR = dataR.GetInt32(1);
                            String firstName = dataR.GetString(2);
                            String lastName = dataR.GetString(3);
                            long noTickets = dataR.GetInt64(4);

                            Buyer buyer = new Buyer(idAR, firstName, lastName, noTickets)
                            {
                                Id = idd
                            };

                            buyers.Add(buyer);
                        }
                    }
                }
            }
            catch (Exception ex)
            {
                Console.ForegroundColor = ConsoleColor.Red;
                Console.WriteLine("Mesajul erorii este {0}", ex.Message);
            }

            return buyers;
        }


        public Buyer Save(Buyer entity)
        {
            if (entity == null)
                throw new ArgumentException("entity must be not null");

            validator.Validate(entity);
            //if (FindOne(entity.Id) != null)
            //{
            //    return entity;
            //}

            var con = DBUtils.getConnection();
            try
            {
                using (var comm = con.CreateCommand())
                {
                    comm.CommandText = "insert into Sellers  values (@idAR, @firstName, @lastName, @noTickets)";

                    var paramIdAR = comm.CreateParameter();
                    paramIdAR.ParameterName = "@idAR";
                    paramIdAR.Value = entity.IdAR;
                    comm.Parameters.Add(paramIdAR);

                    var paramFName = comm.CreateParameter();
                    paramFName.ParameterName = "@firstName";
                    paramFName.Value = entity.FirstName;
                    comm.Parameters.Add(paramFName);

                    var paramLName = comm.CreateParameter();
                    paramLName.ParameterName = "@lastName";
                    paramLName.Value = entity.LastName;
                    comm.Parameters.Add(paramLName);

                    var paramNoTickets = comm.CreateParameter();
                    paramNoTickets.ParameterName = "@noTickets";
                    paramNoTickets.Value = entity.NoTickets;
                    comm.Parameters.Add(paramNoTickets);

                    var result = comm.ExecuteNonQuery();
                    if (result == 0)
                        throw new Exception("Insert failed !");
                    //else
                    //log.InfoFormat("Saved the Buyer {0}", entity);
                }
            }
            catch (Exception ex)
            {
                Console.ForegroundColor = ConsoleColor.Red;
                Console.WriteLine("Mesajul erorii este {0}", ex.Message);
            }

            return null;
        }


        public Buyer Delete(long id)
        {
            if (id == null)
                throw new ArgumentException("Non existent user!");

            var con = DBUtils.getConnection();
            try
            {
                Buyer entity = FindOne(id);

                using (var comm = con.CreateCommand())
                {
                    comm.CommandText = "delete from Buyers where id=@id";
                    IDbDataParameter paramId = comm.CreateParameter();
                    paramId.ParameterName = "@id";
                    paramId.Value = id;
                    comm.Parameters.Add(paramId);
                    var result = comm.ExecuteNonQuery();
                    if (result == 0)
                        return null;
                    else
                    {
                        //log.InfoFormat("Deleted Buyer {0}", entity);
                        return entity;
                    }
                }
            }
            catch (Exception ex)
            {
                Console.ForegroundColor = ConsoleColor.Red;
                Console.WriteLine("Mesajul erorii este {0}", ex.Message);
            }
            return null;
        }

        public Buyer Update(Buyer entity)
        {
            if (entity == null)
                throw new ArgumentException("entity must be not null!");
            validator.Validate(entity);

            if (FindOne(entity.Id) == null)
                return entity;

            var con = DBUtils.getConnection();
            try
            {
                using (var comm = con.CreateCommand())
                {
                    comm.CommandText = "update Buyers set idAR=(@idAR), firstName=(@firstName), lastName=(@lastName), noTickets=(@noTickets) where id=(@id)";

                    var paramIdAR = comm.CreateParameter();
                    paramIdAR.ParameterName = "@idAR";
                    paramIdAR.Value = entity.IdAR;
                    comm.Parameters.Add(paramIdAR);

                    var paramFName = comm.CreateParameter();
                    paramFName.ParameterName = "@firstName";
                    paramFName.Value = entity.FirstName;
                    comm.Parameters.Add(paramFName);

                    var paramLName = comm.CreateParameter();
                    paramLName.ParameterName = "@lastName";
                    paramLName.Value = entity.LastName;
                    comm.Parameters.Add(paramLName);

                    var paramNoTickets = comm.CreateParameter();
                    paramNoTickets.ParameterName = "@noTickets";
                    paramNoTickets.Value = entity.NoTickets;
                    comm.Parameters.Add(paramNoTickets);

                    var paramId = comm.CreateParameter();
                    paramId.ParameterName = "@id";
                    paramId.Value = entity.Id;
                    comm.Parameters.Add(paramId);

                    var result = comm.ExecuteNonQuery();
                    if (result == 0)
                        return entity;
                    else
                    {
                        //log.InfoFormat("Updated Buyer {0}", entity);
                        return null;
                    }
                }
            }
            catch (Exception ex)
            {
                Console.ForegroundColor = ConsoleColor.Red;
                Console.WriteLine("Mesajul erorii este {0}", ex.Message);
            }
            return entity;
        }

    }
}

File: ticketProject8-ORM/TS-ORM/TicketPersistance/src/main/java/ticket/persistance/jdbc/BuyerDatabase.java
package ticket.persistance.jdbc;


import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import ticket.model.Buyer;
import ticket.model.validators.ValidationException;
import ticket.model.validators.Validator;
import ticket.persistance.BuyerRepository;
import ticket.persistance.jdbc.utils.JdbcUtils;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class BuyerDatabase implements BuyerRepository {

    private final JdbcUtils dbUtils;
    final private Validator<Buyer> validator;
    private static final Logger logger= LogManager.getLogger();

    public BuyerDatabase(Validator<Buyer> validator, Properties props) {
        this.validator = validator;
        dbUtils=new JdbcUtils(props);
        logger.info("Initializing BuyerDatabase");
    }

    public Buyer extractEntity(List<String> attributes) {
        if (attributes.size() < 5)
            throw new ValidationException("Insuficient datas!\n");
        Buyer buyer = new Buyer(Long.parseLong(attributes.get(1)), attributes.get(2),attributes.get(3), Long.parseLong(attributes.get(4)));
        buyer.setId(Long.parseLong(attributes.get(0)));
        return buyer;
    }

    protected String createEntityAsString(Buyer entity) {
        return entity.getId() + ";" + entity.getIdAR()+ ";" + entity.getFirstName() + ";" + entity.getLastName() + ";" + entity.getNoTickets();
    }

    @Override
    public Buyer findOne(Long id) {
        if (id == null)
            throw new IllegalArgumentException("id must be not null");

        Connection con= dbUtils.getConnection();
        try {
            PreparedStatement preStmt = con.prepareStatement("select * from Buyers where id = ?");
            preStmt.setInt(1, Math.toIntExact(id));
            ResultSet result = preStmt.executeQuery();

            if (result.next()) {
                Long idd = result.getLong("id");
                Long idAR = result.getLong("idAR");
                String firstName = result.getString("firstName");
                String lastName = result.getString("lastName");
                Long noTickets = result.getLong("noTickets");

                Buyer buyer = new Buyer(idAR, firstName, lastName, noTickets);
                buyer.setId(idd);

                result.close();
                preStmt.close();

                return buyer;
            }
        } catch (Exception e) {
            System.out.println(e.toString());
        }
        return null;
    }

    @Override
    public Iterable<Buyer> findAll() {
        Connection con= dbUtils.getConnection();
        List<Buyer> buyers= new ArrayList<>();

        try{
            PreparedStatement preStmt = con.prepareStatement("select * from Buyers ");
            ResultSet result = preStmt.executeQuery();
            while(result.next()){
                Long id=result.getLong("id");
                Long idAR=result.getLong("idAR");
                String firstName = result.getString("firstName");
                String lastName = result.getString("lastName");
                Long noTickets = result.getLong("noTickets");

                Buyer buyer = new Buyer(idAR, firstName, lastName, noTickets);
                buyer.setId(id);

                buyers.add(buyer);
            }
            result.close();
            preStmt.close();
        }
        catch(SQLException ex){
            System.err.println("Error DB" + ex);
        }

        return buyers;
    }


    @Override
    public Buyer save(Buyer entity) {
        logger.traceEntry("saving Buyer {}", entity);
        if (entity == null)
            throw new IllegalArgumentException("entity must be not null");
        validator.validate(entity);

        Connection con= dbUtils.getConnection();
        try{
            PreparedStatement preStmt = con.prepareStatement("insert into Buyers (idAR, firstName, lastName, noTickets) values (?,?,?,?)");
            preStmt.setInt(1, Math.toIntExact(entity.getIdAR()));
            preStmt.setString(2, entity.getFirstName());
            preStmt.setString(3,entity.getLastName());
            preStmt.setInt(4, Math.toIntExact(entity.getNoTickets()));
            int result=preStmt.executeUpdate();

            preStmt.close();

            logger.trace("Saved {} instances", result);
            if (result == 0)
                throw new Exception("Insert Failed");
        }
        catch(SQLException ex){
            logger.error(ex);
            System.err.println("Error DB" + ex);
        }
        catch(Exception ex){
            System.err.println(ex);
        }

        logger.traceExit();
        return null;
    }

    @Override
    public Buyer delete(Long id) {
        logger.traceEntry("deleting Buyer with id {}", id);
        if (id == null)
            throw new IllegalArgumentException("Non existent user!");

        Connection con= dbUtils.getConnection();
        try {
            Buyer entity = findOne(id);

            PreparedStatement preStmt = con.prepareStatement("DELETE FROM Buyers WHERE id = ?");
            preStmt.setInt(1, Math.toIntExact(id));
            int result = preStmt.executeUpdate();

            preStmt.close();

            //con.close();
            logger.trace("Deleted {} instances", result);
            if (result > 0)
                return entity;
            else
                return null;

        } catch (Exception e) {
            logger.error(e);
            System.out.println(e.toString());
        }

        logger.traceExit();
        return null;
    }

    @Override
    public Buyer update(Buyer entity) {
        logger.traceEntry("updating Buyer {}", entity);
        if (entity == null)
            throw new IllegalArgumentException("entity must be not null!");
        validator.validate(entity);

        if (findOne(entity.getId()) == null)
            return entity;
        Connection con= dbUtils.getConnection();
        try {
            PreparedStatement preStmt = con.prepareStatement("update Buyers set idAR=(?), firstName=(?), lastName=(?), noTickets=(?) where id=(?);");
            preStmt.setInt(1, Math.toIntExact(entity.getIdAR()));
            preStmt.setString(2, entity.getFirstName());
            preStmt.setString(3, entity.getLastName());
            preStmt.setInt(4, Math.toIntExact(entity.getNoTickets()));
            preStmt.setInt(5, Math.toIntExact(entity.getId()));
            int result = preStmt.executeUpdate();

            preStmt.close();

            logger.trace("Updated {} instances", result);
            if (result > 0)
                return null;
            else
                return entity;
        } catch (Exception e) {
            logger.error(e);
            System.out.println(e.toString());
        }

        logger.traceExit();
        return entity;
    }


}
